### PR TITLE
Make it possible to resolve svg's outside \OC::$SERVERROOT

### PR DESCRIPTION
**Pull request by kesselb**
Resolve update blocker for snap.

Is this pr backportable? Alternatively we could keep the current approach and add the patch as fallback.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by rullzer** [comment]
I don't get what this tries to solve...

**Comment by kesselb** [comment]
> I don't get what this tries to solve...

Yes description is missing some details :see_no_evil: 

This PR should fix a issue with resolving the path for svg files for app directories outside the document root. It's related to https://github.com/nextcloud/server/issues/13556 but not a fix for the ResourceLocator warning.

Here are the details: https://github.com/nextcloud/server/issues/13556#issuecomment-557908015

I have no idea what the current approach should do. But `substr($appRootPath, strlen($this->serverRoot))` will not work if the apps directories is not a children of the server root. This will probably break other things :man_shrugging: 

If this patch is accepted it's possible to return svg's also for apps not reachable from the web. That makes no sense but was forbidden before. 

I think the actual issue is somewhere else: https://github.com/nextcloud/server/issues/13556#issuecomment-514827855 For whatever reason symlinks are resolved at some point (probably to make other features work) but that's the actual issue. Nextcloud should work with symlinks.

**Comment by pachulo** [comment]
> Nextcloud should work with symlinks.

I think that this summarizes it quite well: for the snap to work OK, this is required.

Any news on this? Do you need more info @rullzer ?

**Review by nickvergessen — APPROVED** [review]
Didn't test but the change looks good and the unit tests should prove it

File: core/Controller/SvgController.php [hunk 1]
```diff
@@ -97,13 +98,13 @@ public function getSvgFromCore(string $folder, string $fileName, string $color =
...
-
-		if (!$appPath) {
+		try {
+			$appPath = $this->appManager->getAppPath($app);
+		} catch (AppPathNotFoundException $e) {
 			return new NotFoundResponse();
 		}
-		$path = $this->serverRoot . $appPath ."/img/$fileName.svg";
+
+		$path = $appPath . "/img/$fileName.svg";
```
Comment (georgehrke) [inline core/Controller/SvgController.php:107]:
```suggestion
		$path = $appPath . '/img/$fileName.svg';
```

**Review by georgehrke — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@kesselb Backport to stable18?

**Comment by kesselb** [comment]
I would prefer to not backport it ;)

**Pull request merged**